File: programmers/level1/비밀지도.py
def solution(n, arr1, arr2):
    arr = []
    for n1, n2 in zip(arr1, arr2):
        string = ""
        binn1 = bin(n1)[2:]
        binn1 = '0'*(n-len(binn1)) + binn1
        binn2 = bin(n2)[2:]
        binn2 = '0'*(n-len(binn2)) + binn2
        for ch1, ch2 in zip(binn1, binn2):
            if ch1=='1' or ch2=='1': string += '#'
            else: string += ' '
        arr.append(string)
    return arr

File: programmers/level1/비밀지도.java
import java.util.*;

class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] ans = new String[n];
        String bn1;
        String bn2;
        for(int i=0; i<n; i++) {
            StringBuilder sb = new StringBuilder();
            bn1 = Integer.toBinaryString(arr1[i]);
            bn2 = Integer.toBinaryString(arr2[i]);
            for(int j=0; j<n; j++) {
                if(bn1.length()-n+j<0 && bn2.length()-n+j<0) sb.append(" ");
                else if(bn1.length()-n+j<0 && bn2.charAt(bn2.length()-n+j) == '1') sb.append("#");
                else if(bn2.length()-n+j<0 && bn1.charAt(bn1.length()-n+j) == '1') sb.append("#");
                else if(bn1.length()-n+j<0 || bn2.length()-n+j<0) sb.append(" ");
                else if(bn1.charAt(bn1.length()-n+j) == '1' || bn2.charAt(bn2.length()-n+j) == '1') sb.append("#");
                else sb.append(" ");
            }
            ans[i] = sb.toString();
        }
        return ans;
    }
}